fix: LEAP-1931: Correct frame offset calculation in Timeline

Gondragos opened the pull request:
### Reason for change:
The frame offset calculation in the Timeline component did not account for `currentOffsetX`, which led to improper determination of valid offset ranges. This caused inconsistent behavior in seeking the right video frame for displaying.

Before:
![Screen Recording 2025-03-18 at 15 05 56](https://github.com/user-attachments/assets/c6ba3fe5-4bbd-48df-8548-a54cfbe24b72)


### Solution:
The check for frame offsets was adjusted to include `currentOffsetX`. This ensures accurate calculations and resolves the inconsistency.

After:
![Screen Recording 2025-03-18 at 15 22 21](https://github.com/user-attachments/assets/45cd5a33-4781-4a37-ba49-9f96de3e6a4b)


### Rollout strategy:
This code will be rolled out as part of the next release and does not require feature flags or environment variables.

### Testing:
- Manually tested.

### Risks:
This change modifies a critical calculation in the Timeline. There is a minor risk of unforeseen edge cases that might have been missed.

### Reviewer Notes:
From the code perspective `offset` contains the value of the current mouse x coordinate related to the timeline itself plus `currentOffsetX` which is basically `scrollLeft`. Although it's unclear why that method was called `getMouseToFrame`, it's pretty clear that the next condition was working fine only when the timeline's frame is placed at the left edge.

<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

Gondragos committed:
fix: correct frame offset calculation in Timeline

Adjusted the frame offset check to account for `currentOffsetX`, ensuring proper behavior when determining the valid offset range.

Gondragos commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13938272558)
> Successfully merged:  create mode 100644 web/tools/design-tokens-converter/package.json

Gondragos committed:
Merge branch 'develop' into 'fb-LEAP-1931/timelabels-drag-limits'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13938272558

MihajloHoma commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13966532181)
> Successfully merged:  18 files changed, 117 insertions(+), 60 deletions(-)

MihajloHoma committed:
Merge branch 'develop' into 'fb-LEAP-1931/timelabels-drag-limits'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13966532181

MihajloHoma commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13966645534)

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-LEAP-1931/timelabels-drag-limits'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13966645534

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13966645534
